search-app/src/cmr/search/services/acls/collections_cache.clj

```diff
@@ -116,62 +116,59 @@
     (doseq [[coll-key coll-value] collections-map]
       (hash-cache/set-value cache cache-key coll-key (clj-times->time-strs coll-value)))))
 
-;; TODO this is the new func that will replace 'get-collection' AND 'get-collection-map' during transition
-(defn get-collection-gran-acls
- "Gets a single collection from the cache by concept id. Handles refreshing the cache if it is not found in it.
- Also allows provider-id and entry-title to be used."
+(defn get-collection-for-gran-acls
  ([context coll-concept-id]
-  (let [coll-by-concept-id-cache (hash-cache/context->cache context coll-for-gran-acl-caches/coll-by-concept-id-cache-key)
-        collection (hash-cache/get-value
+  "Gets a single collection from the cache by concept id. If collection is not found, will add it to the cache (if it exists in elastic) and will return empty collection."
+ (let [coll-by-concept-id-cache (hash-cache/context->cache context coll-for-gran-acl-caches/coll-by-concept-id-cache-key)
+       collection (hash-cache/get-value
                     coll-by-concept-id-cache
                     coll-for-gran-acl-caches/coll-by-concept-id-cache-key
                     coll-concept-id)]
    (when (or (nil? collection) (empty? collection))
+    ;; if collection not exist in cache, search for it and put in cache
     (info (coll-msg/collection-not-found coll-concept-id))
-    (coll-for-gran-acl-caches/refresh-entire-cache context)) ;;TODO replace
+    (coll-for-gran-acl-caches/set-caches context coll-concept-id))
    (time-strs->clj-times collection)))
  ([context provider-id entry-title]
-  (let [coll-by-provider-id-and-entry-id-cache (hash-cache/context->cache context coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key)
-        collection (hash-cache/get-value
-                    coll-by-provider-id-and-entry-id-cache ;; cache
-                    coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key ;; key
-                    (str provider-id entry-title))]
-   (when (or (nil? collection) (empty? collection))
-    (info (coll-msg/collection-not-found (str provider-id entry-title)))
-    (coll-for-gran-acl-caches/refresh-entire-cache context)) ;;TODO replace
-   (time-strs->clj-times collection))))
+  "Gets a single collection from the cache by concept id. If collection is not found, will add it to the cache (if it exists in elastic) and will return empty collection."
+ (let [coll-by-provider-id-and-entry-title-cache (hash-cache/context->cache context coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key)
+       collection (hash-cache/get-value
+                   coll-by-provider-id-and-entry-title-cache
+                   coll-for-gran-acl-caches/coll-by-provider-id-and-entry-title-cache-key
+                   (str provider-id entry-title))]
+  (time-strs->clj-times collection))))
```

Comment:
Shouldn't you try and load the collection if it doesn't exist in the cache just like on the removed line 139?